frame/bridge/src/lib.rs

```diff
@@ -136,67 +139,84 @@ impl<T: Trait> Module<T> {
 		state_root: &T::Hash,
 		proof: StorageProof,
 		validator_set: &Vec<(AuthorityId, AuthorityWeight)>,
-	) -> Result<(), Error> {
+	) -> DispatchResult {
 
 		let checker = <StorageProofChecker<<T::Hashing as sp_runtime::traits::Hash>::Hasher>>::new(
 			*state_root,
 			proof.clone()
-		)?;
+		);
+
+		// TODO: Remove once I can impl From<StorageError> for DispatchError
```

Comment:
Ideally I would've like to have implemented this, but with the way `decl_error!` is written at the moment I'm not able to do this. If anybody has any better ideas than the `map_storage_err` function I'm using at the moment I'm all ears.

Reply:
Should be just
```rust
let checker = checker.map_err(Self::map_storage_err)?;
```
I don't think that it's so bad that it requires a `TODO` line in the code. Feel free to create an issue for this though.